content/articles/how-to-implement-k-fold-cross-validation/index.md

```diff
@@ -0,0 +1,417 @@
+
+![5 Fold Cross-Validation](/engineering-education/how-to-implement-k-fold-cross-validation/5_fold_cv.jpeg)
+
+[Image Source: Great Learning Blog](https://www.mygreatlearning.com/blog/cross-validation/)
+
+In the first iteration, we use the first part of the data for validation. We use the other parts of the data set for training as illustrated in the image above.
+
+### Data Preprocessing
+ We import all the relevant libraries for the project and load the data set.
+
+```python
+# import relevant libraries
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+from sklearn.metrics import f1_score
+%matplotlib inline
+
+# load dataset
+dataset = pd.read_csv('breast_cancer_data.csv')
+dataset
+```
+
+![Dataset Preview](/engineering-education/how-to-implement-k-fold-cross-validation/dataset_preview.PNG)
+
+The target variable is the diagnosis column. It has an index of 1. The features are all the columns except the id, diagnosis and Unnamed: 32 columns.
+
+```python
+# Separate features and target variable
+X = dataset.iloc[:, 2:-1].values
+y = dataset. iloc [:, 1].values
+print("Matrix of features", X, sep='\n')
+print("--------------------------------------------------")
+print("Target Variable", y, sep='\n')
+```
+
+**Output:**
+```
+    Matrix of features
+    [[1.799e+01 1.038e+01 1.228e+02 ... 2.654e-01 4.601e-01 1.189e-01]
+     [2.057e+01 1.777e+01 1.329e+02 ... 1.860e-01 2.750e-01 8.902e-02]
+     [1.969e+01 2.125e+01 1.300e+02 ... 2.430e-01 3.613e-01 8.758e-02]
+     ...
+     [1.660e+01 2.808e+01 1.083e+02 ... 1.418e-01 2.218e-01 7.820e-02]
+     [2.060e+01 2.933e+01 1.401e+02 ... 2.650e-01 4.087e-01 1.240e-01]
+     [7.760e+00 2.454e+01 4.792e+01 ... 0.000e+00 2.871e-01 7.039e-02]]
+    --------------------------------------------------
+    Target Variable
+    ['M' 'M' 'M' 'M' 'M' 'M' 'M' 'M' 'M' 'M' 'M' 'M' 'M' 'M' 'M' 'M' 'M' 'M'
+     'M' 'B' 'B' 'B' 'M' 'M' 'M' 'M' 'M' 'M' 'M' 'M' 'M' 'M' 'M' 'M' 'M' 'M'
+     'M' 'B' 'M' 'M' 'M' 'M' 'M' 'M' 'M' 'M' 'B' 'M' 'B' 'B' 'B' 'B' 'B' 'M'
+     'M' 'B' 'M' 'M' 'B' 'B' 'B' 'B' 'M' 'B' 'M' 'M' 'B' 'B' 'B' 'B' 'M' 'B'
+     'M' 'M' 'B' 'M' 'B' 'M' 'M' 'B' 'B' 'B' 'M' 'M' 'B' 'M' 'M' 'M' 'B' 'B'
+     'B' 'M' 'B' 'B' 'M' 'M' 'B' 'B' 'B' 'M' 'M' 'B' 'B' 'B' 'B' 'M' 'B' 'B'
+     'M' 'B' 'B' 'B' 'B' 'B' 'B' 'B' 'B' 'M' 'M' 'M' 'B' 'M' 'M' 'B' 'B' 'B'
+     'M' 'M' 'B' 'M' 'B' 'M' 'M' 'B' 'M' 'M' 'B' 'B' 'M' 'B' 'B' 'M' 'B' 'B'
+     'B' 'B' 'M' 'B' 'B' 'B' 'B' 'B' 'B' 'B' 'B' 'B' 'M' 'B' 'B' 'B' 'B' 'M'
+     'M' 'B' 'M' 'B' 'B' 'M' 'M' 'B' 'B' 'M' 'M' 'B' 'B' 'B' 'B' 'M' 'B' 'B'
+     'M' 'M' 'M' 'B' 'M' 'B' 'M' 'B' 'B' 'B' 'M' 'B' 'B' 'M' 'M' 'B' 'M' 'M'
+     'M' 'M' 'B' 'M' 'M' 'M' 'B' 'M' 'B' 'M' 'B' 'B' 'M' 'B' 'M' 'M' 'M' 'M'
```

Comment:
All your outputs need to be inside a `bash` command. Go through the whole tutorial and edit accordingly

Reply:
done